gateway/api/user.go

```
@@ -58,10 +58,25 @@ func (u *UserRouter) createUser(w http.ResponseWriter, r *http.Request) {
 }
 
 func (u *UserRouter) login(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusOK)
+	email := r.FormValue("email")
+	password := r.FormValue("password")
+	if email == "" || password == "" {
+		render.Render(w, r, res.ErrBadRequest(r, errors.New("missing fields"), ""))
+		return
+	}
 
-	// TODO: implement
+	resp, err := u.c.Login(context.Background(), &request.Login{Email: email, Password: password})
+	if err != nil {
+		render.Render(w, r, res.ErrInternalServer(r, err))
+		return
+	}
+	if resp.GetMessage() != "user successfully logged in" {
```

see other comment about unsuccessful login - I think we should return an error, and then here inspect the contents of that error, since I feel non-error response implies a success as far as grpc goes